Symfony/CS/Tests/ProjectCodeTest.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/*
+ * This file is part of the PHP CS utility.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Tests;
+
+use Symfony\Component\Finder\Finder;
+
+/**
+ * @author Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * @internal
+ */
+final class ProjectCodeTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @dataProvider provideTestClasses
+     */
+    public function testThatTestClassesAreAbstractOrFinal(\ReflectionClass $rc)
+    {
+        $this->assertTrue(
+            $rc->isAbstract() || $rc->isFinal(),
+            sprintf('Test class %s should be abstract or final.', $rc->getName())
+        );
+    }
+
+    public function provideTestClasses()
+    {
+        return array_map(
+            function ($item) {
+                return array(new \ReflectionClass($item));
+            },
+            $this->getClasses('Symfony\\CS\\Tests\\')
+        );
+    }
+
+    private function getClasses($prefix, array $classes = null)
+    {
+        static $projectClasses = null;
+
+        if (null === $classes && null === $projectClasses) {
+            $this->registerAllProjectClasses();
+            $projectClasses = $this->getClasses('Symfony\\CS\\', get_declared_classes());
+        }
+
+        if (null === $classes) {
+            $classes = $projectClasses;
+        }
+
+        $ns = $prefix;
+        $prefixLength = strlen($prefix);
+
+        return array_filter(
+            $classes,
+            function ($item) use ($prefix, $prefixLength) {
+                return $prefix === substr($item, 0, $prefixLength);
+            }
+        );
+    }
+
+    private function registerAllProjectClasses()
+    {
+        $finder = Finder::create()
+            ->files()
+            ->in(__DIR__.'/..')
+            ->exclude(array(
```

Comment:
I would add `->name('*.php')` as well, so `readme`, `.php_cs`, etc. can still be added later on into the directories (when needed).